File: Frontend/src/features/courses/types/ModuleDto.ts
import type { LinkDto } from "@/types/LinkDto";

/**
 * Backend DTO: Represents a lesson within a module
 */
export interface ModuleLessonDto {
  lessonId: string;
  title: string;
  index: number;
  duration: string;
  thumbnailUrl: string | null;
  access: string;
  links: LinkDto[];
}

/**
 * Backend DTO: Represents a course module with its lessons
 */
export interface ModuleDto {
  id: string;
  title: string;
  index: number;
  lessonCount: number;
  duration: string;
  lessons: ModuleLessonDto[];
  links: LinkDto[];
}

File: src/CourseService/Application/Courses/Queries/GetById/ModuleDto.cs
using Courses.Application.Services.LinkProvider.Abstractions;

namespace Courses.Application.Courses.Queries.GetById;

public record ModuleDto(
    Guid Id,
    string Title,
    int Index,
    int LessonCount,
    TimeSpan Duration,
    IReadOnlyList<LessonDto> Lessons,
    IReadOnlyList<LinkDto> Links
);
